cookbooks/bcpc-hadoop/attributes/monitoring.rb

```diff
@@ -0,0 +1,121 @@
+default["bcpc"]["hadoop"]["zabbix"]["cron_check_time"] = 240
+default["bcpc"]["hadoop"]["zabbix"]["mail_source"] = "zabbix.zbx_mail.sh.erb"
+default["bcpc"]["hadoop"]["zabbix"]["cookbook"] = nil 
+
+# Graphite queries which specify property to query and alarming trigger, severity(maps to zabbix's api -> trigger -> priority)
+# and owner who the trigger is routed to for resolution
+# Queries are structured as they appear in graphite dashboard
+# Refer Zabbix API(https://www.zabbix.com/documentation/2.2/manual/api/reference/) for the various attributes used
+# The following list is supposed to be a skeleton which should be extended/edited according to the deployment scenario
+# By default items, triggers actions are disabled and won't be monitored. To enable them, set "enable=true" in the json structure
+default["bcpc"]["hadoop"]["graphite"]["queries"] = {
+   'namenode' => {
+      'nnheapmem' => {
+         'type' => "jmx",
+         'query' => "memory.HeapMemoryUsage_committed",
+         'trigger_val' => "max(61,0)",
+         'value_type' => 3,
+         'trigger_cond' => "=0",
+         'trigger_name' => "NameNodeAvailability",
+         'enable' => false,
+         'trigger_desc' => "Namenode service seems to be down",
+         'severity' => 5,
+         'route_to' => "admin"
+      },
+      'numstaledn' => {
+         'type' => "jmx",
+         'query' => "nn_fs_name_system_state.FSNamesystemState.NumStaleDataNodes",
+         'enable' => false
+      }
+   },
+   'hbase_master' => {
+      'hbasenonheapmem' => {
+         'type' => "jmx",
+         'query' => "memory.NonHeapMemoryUsage_committed",
+         'trigger_val' => "max(61,0)",
+         'value_type' => 3,
+         'trigger_cond' => "=0",
+         'trigger_name' => "HBaseMasterAvailability",
+         'enable' => false,
+         'trigger_dep' => ["NameNodeAvailability"],
+         'trigger_desc' => "HBase master seems to be down",
+         'severity' => 5,
+         'route_to' => "admin"
+      },
+      'hbaseheapmem' => {
+         'type' => "jmx",
+         'query' => "memory.HeapMemoryUsage_committed",
+         'history_days' => 2,
+         'trend_days' => 30,
+         'enable' => false
+      },
+      'numrsservers' => {
+         'type' => "jmx",
+         'query' => "hbm_server.Master.numRegionServers",
+         'trigger_val' => "max(61,0)",
+         'value_type' => 3,
+         'trigger_cond' => "=0",
+         'trigger_name' => "HBaseRSAvailability",
+         'enable' => false,
+         'trigger_dep' => ["HBaseMasterAvailability"],
+         'trigger_desc' => "HBase region server seems to be down",
+         'severity' => 5,
+         'route_to' => "admin"
+      }
+   },
+   'bcpc-vm' => {
+      # Entry in suffix array added to hostname as suffix and 
+      # to key name, trigger name as suffix "_<hostname with suffix>"
+      # If specified, should be *first* entry for a host
+      'suffix' =>  [ "1", "2", "3" ], 
+      'memory_active' => {
```

Comment:
[value_type](https://www.zabbix.com/documentation/2.2/manual/api/reference/item/object) is `Type of information of item`. [Default value](https://github.com/kiiranh/chef-bach/blob/monitoring/cookbooks/bcpc-hadoop/recipes/graphite_to_zabbix.rb#L98-L102) is set to *3 (numeric unsigned)*